addon/webextension/selector/ui.js

```diff
@@ -198,83 +201,193 @@ window.ui = (function () { // eslint-disable-line no-unused-vars
+              this.document.documentElement
+            );
+            installHandlerOnDocument(this.document);
+            if (this.addClassName) {
+              this.document.body.className = this.addClassName;
+            }
+            const overlay = this.document.querySelector(".preview-overlay");
+            overlay.querySelector(".preview-instructions").textContent = browser.i18n.getMessage("screenshotInstructions");
+            overlay.querySelector(".myshots-text").textContent = browser.i18n.getMessage("myShotsLink");
+            overlay.querySelector(".visible").textContent = browser.i18n.getMessage("saveScreenshotVisibleArea");
+            overlay.querySelector(".full-page").textContent = browser.i18n.getMessage("saveScreenshotFullPage");
+            overlay.querySelector(".myshots-button").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onOpenMyShots), false);
+            overlay.querySelector(".visible").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onClickVisible), false);
+            overlay.querySelector(".full-page").addEventListener(
+              "click", watchFunction(standardOverlayCallbacks.onClickFullPage), false);
+            resolve();
+          });
+          document.body.appendChild(this.element);
+          this.unhide();
+        } else {
+          resolve();
+        }
+      });
     },
 
-    resetPosition: function () {
-      if (! this.movingEl) {
-        return;
+    updateElementSize: function (force) {
+      // Note: if someone sizes down the page, then the iframe will keep the
+      // document from naturally shrinking.  We use force to temporarily hide
+      // the element so that we can tell if the document shrinks
+      const visible = this.element.style.display !== "none";
+      if (force && visible) {
+        this.element.style.display = "none";
+      }
+      let height = Math.max(
+        document.documentElement.clientHeight,
+        window.innerHeight);
+      if (height !== this.sizeTracking.lastHeight) {
+        this.sizeTracking.lastHeight = height;
+        this.element.style.height = height + "px";
+      }
+      let width = Math.max(
+        document.documentElement.clientWidth,
+        window.innerWidth);
+      if (width !== this.sizeTracking.lastWidth) {
+        this.sizeTracking.lastWidth = width;
+        this.element.style.width = width + "px";
+      }
+      if (force && visible) {
+        this.element.style.display = "";
       }
-      let scrollX = window.scrollX;
-      let scrollY = window.scrollY;
-      this.movingEl.style.top = scrollY + "px";
-      this.movingEl.style.left = scrollX + "px";
-      this.movingEl.style.width = window.innerWidth + "px";
-      this.movingEl.style.height = window.innerHeight + "px";
+    },
+
+    hide: function () {
+      window.removeEventListener("scroll", this.onScroll, false);
```

Comment:
The hoverbox remains visible during the scroll without this. I think it looks bad, but if that's not an issue I'll remove it.